core/container/src/main/java/org/wildfly/swarm/Swarm.java

```diff
@@ -464,7 +464,13 @@ private static boolean isFatJar() throws IOException {
 
     private void loadStageConfiguration(URL url) {
         List<ProjectStage> projectStages = new ProjectStageFactory().loadStages(url);
-        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE, "default");
+        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE);
```

Comment:
I like the if/else approach but it seems that in Java 8 it is better to use Optionals.

```
Optional.ofNullable(System.getenv(PROJECT_STAGE))
                .orElse(Optional.ofNullable(System.getProperty(PROJECT_STAGE))
                .orElse("default"));
```